pagopa/io-app
# chore(Cross):[IOAPPX-422,IOCOM-2065] Replace the legacy markdown renderer with IOMarkdown

## Short description
This PR removes any legacy markdown renderer in favour of IOMarkdown

## List of changes proposed in this pull request
- Remove `LegacyMarkdown`
 
### Preview

<details><summary><code>CiePinScreen</code></summary>
<p>

 |Old|New|
 |-|-|
|![Old](https://github.com/user-attachments/assets/1b0a9e5e-855a-43aa-b118-31b44591701f)|![New](https://github.com/user-attachments/assets/7b565de4-29ac-42e6-b5be-3808b719dc32)|

</p>
</details> 

<details><summary><code>FciDocumentsScreen</code></summary>
<p>

 |Old|New|
 |-|-|
|![FciDocumentsScreen_old](https://github.com/user-attachments/assets/132fc6aa-03ee-4735-b60b-4eb066561b24)|![FciDocumentsScreen_new](https://github.com/user-attachments/assets/bd3433db-4e80-436f-83d0-d758c7b32b98)|

</p>
</details>

<details><summary><code>EycaInformationComponent</code></summary>
<p>

 |Old|New|
 |-|-|
|![EycaInformationComponent_old](https://github.com/user-attachments/assets/8ff1c974-4619-4a99-96ee-5e60ae761793)|![EycaInformationComponent_new](https://github.com/user-attachments/assets/abbb1221-4c9d-4413-8cdd-8a0a79eca702)|

</p>
</details>

<details><summary><code>InitiativeDetailsScreen</code></summary>
<p>

|Old|New|New|
|-|-|-|
|![InitiativeDetailsScreen_old](https://github.com/user-attachments/assets/a2739339-c64d-4c72-bbb6-99ed13d2b7ac)|![InitiativeDetailsScreen_new_0](https://github.com/user-attachments/assets/ff90f8a9-bb65-43ba-9516-234072cc8467)|![InitiativeDetailsScreen_new_1](https://github.com/user-attachments/assets/a35806fb-2371-4c31-a80e-629ce7279ffd)|

</p>
</details>

<details><summary><code>RootedDeviceModal</code></summary>
<p>

 |Old|New|
 |-|-|
 |![RootedDeviceModal_old](https://github.com/user-attachments/assets/903c6296-3f93-4bd7-87b2-215c7570ea2b)|![RootedDeviceModal_new](https://github.com/user-attachments/assets/cbab64e7-46ff-423c-bd9b-9e84fa3efe82)|
 
 </p>
</details>

<details><summary><code>ShareDataFeatureInfos</code></summary>
<p>

 |Old|New|
 |-|-|
 |![Simulator Screenshot - iPhone 16 Pro - 2024-11-26 at 13 44 06](https://github.com/user-attachments/assets/eb374795-4096-420c-8636-9c28bebaa0aa) | ![Simulator Screenshot - iPhone 16 Pro - 2024-11-26 at 13 48 00](https://github.com/user-attachments/assets/04624c68-f16e-4e5a-ab6b-f2c0ec09d9b9) |

 </p>
</details>

<details><summary><code>ManualConfigBottomSheet</code></summary>
<p>

 |Old|New|
 |-|-|
 | ![Simulator Screenshot - iPhone 16 Pro - 2024-11-26 at 13 44 53](https://github.com/user-attachments/assets/3bb687a5-d440-46e3-8b80-947de4181e86) | ![Simulator Screenshot - iPhone 16 Pro - 2024-11-26 at 13 48 13](https://github.com/user-attachments/assets/6a08f1f0-76e7-4a33-b6dd-7a9330513d1c) |

 </p>
</details>

<details><summary><code>InitiativeRulesInfoBox</code></summary>
<p>

 |Old|New|
 |-|-|
 | ![InitiativeRulesInfoBox_old](https://github.com/user-attachments/assets/8d12068c-971e-464f-a12d-b41fd1a20264)|![InitiativeRulesInfoBox_new](https://github.com/user-attachments/assets/73c88640-31ea-4394-becd-f7a6bc79a5ad)|

 </p>
</details>

<details><summary><code>PDNDPrerequisitesScreen</code></summary>
<p>

 |Old|New|
 |-|-|
|![PDNDPrerequisitesScreen_old](https://github.com/user-attachments/assets/4d11211d-486b-42ba-bcba-e73f5abd1c82)|![PDNDPrerequisitesScreen_new](https://github.com/user-attachments/assets/6b1af002-ecf9-4cc0-b4ac-2e89ed1baf8d)|

 </p>
</details>

<details><summary><code>MessageDetailsScreen</code></summary>
<p>

 |Old|New|
 |-|-|
|![M001_old](https://github.com/user-attachments/assets/900d6ef4-de5e-42dd-9938-63483d4da0df)|![M001_new](https://github.com/user-attachments/assets/6dbd5c7a-acd0-4c56-b819-1a9ef3ce6c0f)|
|![M002_old](https://github.com/user-attachments/assets/9c556b41-d576-4dcb-a628-51551c32ff27)|![M002_new](https://github.com/user-attachments/assets/3498a4ad-5d77-4697-8728-5e8bbac14978)|
|![M003_old](https://github.com/user-attachments/assets/70dd705e-5aae-47ea-b1b7-1cf2551ddbd3)|![M003_new](https://github.com/user-attachments/assets/8ffc3654-3bcd-4315-a420-d6baa7567e5b)|
|![M004_old](https://github.com/user-attachments/assets/369d2661-eb07-4de8-8e5f-fcf35101ad7e)|![M004_new](https://github.com/user-attachments/assets/e2e48a62-2158-4c24-8ce5-b3e0b134bf62)|
|![M005_old](https://github.com/user-attachments/assets/fd8a5ea5-828e-444f-9c63-d7e3b0b7f709)|![M005_new](https://github.com/user-attachments/assets/b4fd2d00-f3b2-4ca9-b5e9-98afa54dd91f)|

 </p>
</details>

<details><summary><code>ServiceDetailsScreen</code></summary>
<p>

 |Old|New|
 |-|-|
|![S001_old](https://github.com/user-attachments/assets/ceea38be-57a8-4109-922f-a4bbff7cf2f8)|![S001_new](https://github.com/user-attachments/assets/65989417-9abf-4e92-86a7-2d594e8d24ce)|

 </p>
</details>

<details><summary><code>CgnInformationScreen</code></summary>
<p>

 |Old|New|
 |-|-|
|![CGN001_old](https://github.com/user-attachments/assets/82f3e188-8e15-45fe-878e-281174467184)|![cgn001](https://github.com/user-attachments/assets/a9a25e5c-9060-4c25-abd1-48664369fa36)|
|![CGN002_old](https://github.com/user-attachments/assets/973a2847-14ae-4292-b330-a9ad2ff009a9)|![cgn002](https://github.com/user-attachments/assets/c8980967-dcd2-4977-b2a3-cca25da99a8e)|
|![CGN003_old](https://github.com/user-attachments/assets/39f58c34-93e8-44a1-9edc-24199bf16deb)|![cgn003](https://github.com/user-attachments/assets/e56167fc-6a63-4c45-8934-ff0e00369627)|
|![CGN004_old](https://github.com/user-attachments/assets/70a03b46-acc8-41df-9a42-43e3d5141b4c)|![cgn004](https://github.com/user-attachments/assets/afe4b690-80aa-4575-abc8-5af42867d0d9)|

 </p>

</details>

### Authors
@Vangaorth @adelloste @dmnplb 

## How to test
All affected sections have been tested and related screenshots are linked in this PR. Assuming that IOMarkdown is a fully working component, either statically check that the edited code makes sense or, using the io-dev-api-server, verify that every impacted screen/component/bottom-sheet renders properly
